pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -978,6 +978,7 @@ private void closeStreamConsumers() {
 
   private void closePartitionGroupConsumer() {
     try {
+      _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
```

Comment:
Why not modify the interface to have a `close(offset)` call ? that way, underlying streams can do whatever they want with the offset.

Also, it mirrors how the consumer is constructed with the start offset

Reply:
1. I would like to avoid interface changes for bugfixes. This is a bug that is valid only after the introduction of checkpoint interface.
2. It is not just the underlying streams that need to handle this. The LLRealtimeSegmentDataManager should also use the current offset when creating a new consumer and not always the startOffset.